packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -351,6 +351,15 @@ export function analyze_component(root, source, options) {
 				}
 			}
 
+			// if we are creating a synthetic binding for a let declaration we should also declare
+			// the declaration as state in case it's reassigned
+			if (
+				declaration !== null &&
+				declaration.kind === 'normal' &&
+				declaration.declaration_kind === 'let'
```

Comment:
Mmm let me check...the original issue was about the reassignment and that's why i wrote the comment...if normally it's not a problem i guess we can add the extra condition

Reply:
Yeah i guess the extra condition doesn't hurt (it will still be state tho if it's never reassigned and has an init but this should filter out the case when it's not assigned and it doesn't have an init...which will be a runtime error anyway :smile: )